src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -275,24 +277,65 @@ protected override sealed void ProcessRecord()
         #region Cmdlets base APIs
 
         /// <summary>
-        /// Initializes source repositories for PowerShell cmdlets, based on config, source string, and/or host active source property value.
+        /// Looks through all available sources (including those disabled) by matching source name and url to get matching sources.
         /// </summary>
         /// <param name="source">The source string specified by -Source switch.</param>
-        protected void UpdateActiveSourceRepository(string source)
+        /// <returns></returns>
+        protected PackageSource GetMatchingSource(string source)
         {
             var packageSources = _sourceRepositoryProvider?.PackageSourceProvider?.LoadPackageSources();
+            var matchingSource = packageSources
+                ?.Where(p => StringComparer.OrdinalIgnoreCase.Equals(p.Name, source) ||
+                             StringComparer.OrdinalIgnoreCase.Equals(p.Source, source))
+                .FirstOrDefault();
+
+            return matchingSource;
+        }
 
+        /// <summary>
+        /// Checks if the sourse is valid http, local or known source. Else throws an exception.
+        /// </summary>
+        /// <param name="source">The source string specified by -Source switch.</param>
+        /// <param name="packageId">The package ID string specified.</param>
+        /// <param name="matchingSource">Matching Source if the specified source is a known source. Else null.</param>
+        /// <returns></returns>
+        protected string CheckSourceValidity(string source, string packageId, PackageSource matchingSource)
```

Comment:
Let the caller check if `matchingSource` is not null before calling the method. Then remove it from argument list and simplify the signature.